Source/AwsCommonRuntimeKit/sdkutils/FileBasedConfiguration.swift

```diff
@@ -0,0 +1,182 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+
+import AwsCSdkUtils
+import struct Foundation.Data
+
+public class FileBasedConfiguration {
+    var rawValue: OpaquePointer
+
+    /// If the `AWS_PROFILE` environment variable is set, use it. Otherwise, return "default".
+    public static var defaultProfileName: String {
+        let profileName = aws_get_profile_name(defaultAllocator.rawValue, nil)!
+        defer {
+            aws_string_destroy(profileName)
+        }
+        return String(awsString: profileName)!
+    }
```

Comment:
Using a defaultAllocator directly allows it to be made a variable instead of a function. As discussed offline, we will remove support for per-api allocator overriding functionality in a future PR.